### PR TITLE
WIP: Support PostgreSQL Databases

### PR DESCRIPTION
**Pull request by carsonab**
Adds support for PostgreSQL:
- PostgreSQL database connections with TLS
  - Local testing can be done against a `alloydbomni` container
  - Added `gencerts.sh` to create certs used in local server config and client tests to enable TLS tests
- AlloyDB connections using [alloybd-go-connector](https://github.com/GoogleCloudPlatform/alloydb-go-connector)
- Uses single database driver `pgx`

Local testing can be done against a `alloydbomni` container, which uses the same database engine as hosted AlloyDB, using a normal postgres connection. Connections with a alloydb-go-connector need a GCP AlloyDB instance to connect to. However both connection methods use the same database driver. This means our local testing of business logic and its sql queries use the same database driver and engine as higher environments, and only the connection method will differ between environments.

Tested with alloybd-go-connector successfully using hosted AlloyDB instance for both IAM and native db login.

TODOs:
- handle closing dialer from `alloydbconn.NewDialer`

NOTE:
- Tested client certs by manually editing the server's `pg_hba.conf` to require client certs with `hostssl all all all scram-sha-256 clientcert=verify-full` but not automated to do that

### CODE REVIEW

**Comment by adamdecaf** [comment]
Do we want to start this new database work in moovfinancial rather than in here?

File: database/migrator.go [hunk 1]
```diff
@@ -155,6 +156,29 @@ func getDriver(logger log.Logger, config DatabaseConfig, opts *migrateOptions) (
 				return nil, nil, err
 			}
 		}
+	} else if config.Postgres != nil {
+		if opts.source == nil {
+			src, err := NewPkgerSource("postgres", false)
```
Comment (adamdecaf) [inline database/migrator.go:161]: We should be able to remove pkger. We've been using go:embed for quite a while. Can leave it in this PR for now.

File: database/postgres.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		dialer = d
+		dsn = fmt.Sprintf(
+			// sslmode is disabled because the alloy db connection dialer will handle it
+			"user=%s password=%s dbname=%s sslmode=disable",
+			config.User, config.Password, databaseName,
+		)
+	}
+
+	// TODO
+	//cleanup := func() error { return d.Close() }
```
Comment (adamdecaf) [inline database/postgres.go:113]: Needed?

File: database/postgres_test.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	config := database.DatabaseConfig{
+		DatabaseName: "moov",
+		Postgres: &database.PostgresConfig{
+			Address:  "localhost:5432",
+			User:     "moov",
+			Password: "moov",
+			TLS: &database.PostgresTLSConfig{
+				CACertFile:     "../testcerts/root.crt",
+				ClientCertFile: "../testcerts/client.crt",
+				ClientKeyFile:  "../testcerts/client.key",
```
Comment (adamdecaf) [inline database/postgres_test.go:48]: Need to use filepath.Join

File: database/postgres.go (hunk 2 above)
Comment (carsonab) [reply database/postgres.go:113]: Yea, listed this under the TODOs in the description. We need to call this when we're done with the connection

File: database/migrator.go (hunk 1 above)
Comment (InfernoJJ) [reply database/migrator.go:161]: yeah I'll remove that here soon.

**Comment by alovak** [comment]
@carsonab thanks for the update! Looks solid!

**Comment by carsonab** [comment]
Recreated branch on moov-io/base directly with maintainer permissions, closing this PR in favor https://github.com/moov-io/base/pull/437

**Pull request closed without merging**